apps/withdraw-pool/src/utils/decodeHookCalldata.ts

```diff
@@ -27,5 +34,35 @@ export function findPoolIdOnCallData(string: `0x${string}`) {
 
   const lastCallIndex = calls.length - 1;
   const call = calls[lastCallIndex];
-  return call.target.toLowerCase();
+
+  const decodedCoWAmmCallData = decodeFunctionData({
+    abi: cowAmmAbi,
+    data: call.callData as `0x${string}`,
+  });
+
+  const poolAddress = call.target;
+
+  if (decodedCoWAmmCallData.functionName !== "exitPool") {
+    return;
+  }
+
+  if (!decodedCoWAmmCallData.args?.length) {
+    return;
+  }
```

Comment:
should these throw instead? If we're just using this to decode an `exitPool` call, or should this function be called something like `decodeExitPoolHookCalldata` ?